src/System.Threading.Thread/src/System/Threading/Thread.cs

```diff
@@ -155,6 +155,10 @@ public static IPrincipal CurrentPrincipal
         {
             get
             {
+                if (CurrentThread._principal == null)
```

Comment:
`return CurrentThread._principal ?? (CurrentThread._principal = AppDomain.CurrentDomain.GetThreadPrincipal());` may be a bit more efficient. Saves one `CurrentThread._principal` evaluation on the hot path - the JIT is unlikely to optimize it out.